### PR TITLE
added retry functionality - resolves #4

### LINKED ISSUE #4: Add Catch & Retry
Following the existing issue in the original repo (https://github.com/vkkis93/serverless-step-functions-offline/issues/14), we want to implement Catch & Retry as well.

I am not sure how `BackoffRate` and `IntervalSeconds` will work considering all states are executed synchronously. It may be better to defer implementation of those until proper asynchronous task execution can be implemented. This will assist for accurate `Parallel` processing & the `Map`.`MaxConcurrency` feature.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -657,12 +660,43 @@ export default class StepFunctionsOfflinePlugin implements Plugin {
...
         throw `Error in function "${this.currentStateName}": ${JSON.stringify(err)}`;
       }
-      if (!notEmpty(this.currentState)) return;
+      if (err && isType('Task')<Task>(this.currentState)) {
+        const matchingError = (this.currentState.Retry ?? []).find(condition =>
+          condition.ErrorEquals.includes('HandledError')
+        );
+        if (!matchingError) throw `Error in function "${this.currentStateName}": ${JSON.stringify(err)}`;
+        attempt += 1;
+        if (attempt < (matchingError.MaxAttempts ?? 0)) {
```
Comment (mitch-byb) [inline src/index.ts:678]: Does this need to be `<=`?

File: src/index.ts [hunk 2]
```diff
@@ -657,12 +660,43 @@ export default class StepFunctionsOfflinePlugin implements Plugin {
...
+          if (matchingError.IntervalSeconds !== undefined && matchingError.IntervalSeconds !== 0) {
+            const backoffRate = matchingError?.BackoffRate ?? 2;
+            const fullDelay =
+              attempt === 1
+                ? matchingError.IntervalSeconds
+                : matchingError.IntervalSeconds * (attempt - 1) * backoffRate;
+            console.log(`Delaying ${fullDelay} seconds for execution #${attempt + 1} of state ${name}`);
```
Comment (mitch-byb) [inline src/index.ts:685]: Should we use `this.cliLog`?

**Review by mitch-byb — APPROVED** [review]
Looking good

File: src/index.ts (hunk 1 above)
Comment (ghost) [reply src/index.ts:678]: attempt starts at 1, so attempt=5 would be execution=6, we actually want to attempt it 5 times=6 executions

**Review by mitch-byb — APPROVED** [review]
👍

**Pull request merged**